packages/core/src/external/aws/lambda-logic/bundle-to-html-derm.ts

```diff
@@ -0,0 +1,2688 @@
+
+  encounters.forEach(encounter => {
+    if (encounter.diagnosis) {
+      encounter.diagnosis.forEach(diagnosis => {
+        if (diagnosis.condition && diagnosis.condition.reference) {
+          const conditionId = diagnosis.condition.reference.split("/")[1];
+          if (encounter.period && conditionId) {
+            conditionDates[conditionId] = {
+              start: encounter.period.start ?? "",
+              end: encounter.period.end ?? "",
+            };
+          }
+        }
+      });
+    }
+  });
+
+  return conditionDates;
+}
+
+function createSection(title: string, tableContents: string) {
+  return `
+    <div id="${title.toLowerCase().replace(/\s+/g, "-")}" class="section">
+      <div class="section-title">
+        <h3 id="${title}" title="${title}">&#x276F; ${title}</h3>
+        <a href="#mr-header">&#x25B2; Back to Top</a>
+      </div>
+      <div class="section-content">
+          ${tableContents}
+      </div>
+    </div>
+  `;
+}
+
+function mapResourceToId<ResourceType>(resources: Resource[]): Record<string, ResourceType> {
+  return resources?.reduce((acc, resource) => {
+    const id = resource?.id ?? "";
+
+    return {
+      ...acc,
+      [id]: resource,
+    };
+  }, {});
+}
+
+// find condition with code Z00 in the past year
+function getDermConditions(conditions: Condition[]): Condition[] {
+  const dermConditions = conditions.filter(condition => {
+    const code = getSpecificCode(condition.code?.coding ?? [], [ICD_10_CODE]);
+
+    // Check if the code is in the L00-L99 range
+    const isDermCode = code ? /L\d{2}(?:\.\d+)?/i.test(code) : false;
+
+    return isDermCode;
+  });
+
+  return dermConditions;
+}
+
+function getRheumatoidConditions(conditions: Condition[]) {
+  const rheumatoidConditions = conditions.filter(condition => {
+    const code = getSpecificCode(condition.code?.coding ?? [], [ICD_10_CODE]);
+
+    // Check if the code is in the M00-M99 range
+    const isRheumatoidCode = code ? /M\d{2}(?:\.\d+)?/i.test(code) : false;
+
+    return isRheumatoidCode;
+  });
+
+  return rheumatoidConditions;
+}
+
+function getAsthmaConditions(conditions: Condition[]) {
+  const rheumatoidConditions = conditions.filter(condition => {
+    const code = getSpecificCode(condition.code?.coding ?? [], [ICD_10_CODE]);
+
+    // Check if the code is in the L00-L99 range
+    const isAsthmaCode = code ? code.includes("J45") : false;
+
+    return isAsthmaCode;
+  });
+
+  return rheumatoidConditions;
+}
+
+function hasClinicalRelevantData(fhirTypes: FhirTypes): boolean {
+  const hasValues: string[] = [];
+
+  Object.entries(fhirTypes).forEach(([key, value]) => {
+    const isNotRelatedPersons = key !== "relatedPersons";
+    const isNotCoverages = key !== "coverages";
+    const hasValue = value && Array.isArray(value) && value.length;
+
+    if (isNotRelatedPersons && isNotCoverages && hasValue) {
+      hasValues.push(key);
+    }
+  });
```

Comment:
nit: if we make this a simple `for...of`, we can return from the function `hasClinicalRelevantData` without having to go through all items of `fhirTypes`